Modules/_io/textio.c

```diff
@@ -1992,6 +1992,15 @@ _io_TextIOWrapper_read_impl(textio *self, Py_ssize_t n)
         if (bytes == NULL)
             goto fail;
 
+        if (bytes == Py_None){
+            PyErr_SetString(PyExc_BlockingIOError, "Unexpected None encountered. This may be due to non-blocking I/O "
+                                                   "or an issue with the underlying I/O implementation. Please refer to "
```

Comment:
personal nitpick / nit: I don't think the "Please refer to" part here I don't think is needed stylistically, but also don't know the python documentation standards really well